### DISCUSSION TITLE
RGBA support with 'split_sharp_edges' in PBR rendering

### DISCUSSION THREAD

**Discussion by danbodily** [pyvista/pyvista#4344] [Q&A]
I'm trying to import and plot a "PLY" file from Solidworks that has color assigned to it.  When the mesh is imported it comes in with an RGBA array aligned with the number of 'cells' in the mesh (not the number of points).

I'd like to plot this using PBR with the 'split_sharp_edges' options enabled.   However as soon try this I get:
```
  File "tool_import_test.py", line 62, in <module>
    p.add_mesh(tool, **tool_add_mesh_args)
  File "C:\envs\udo_req\lib\site-packages\pyvista\plotting\plotting.py", line 2179, in add_mesh
    mesh, scalars, texture, split_sharp_edges, feature_angle, preference
  File "C:\envs\udo_req\lib\site-packages\pyvista\plotting\_plotting.py", line 88, in prepare_smooth_shading
    mesh.point_data[indices_array] = np.arange(arr_sz, dtype=np.int32)
  File "C:\envs\udo_req\lib\site-packages\pyvista\core\datasetattributes.py", line 227, in __setitem__
    self.set_array(value, name=key)
  File "C:\envs\udo_req\lib\site-packages\pyvista\core\datasetattributes.py", line 613, in set_array
    vtk_arr = self._prepare_array(data, name, deep_copy)
  File "C:\envs\udo_req\lib\site-packages\pyvista\core\datasetattributes.py", line 769, in _prepare_array
    raise ValueError(f'data length of ({data.shape[0]}) != required length ({array_len})')
ValueError: data length of (163714) != required length (82831)
```
It appears it expects the RGBA array to be aligned with the number of POINTS of the mesh not CELLS.

I can fix this by either disabling PBR, or disabling split_sharp_edges but both these options compromise the look of the mesh I'm after.  Is there a way to map an RGBA array to match the size of the points instead of the cells?  I can fix it this way as well.

Perhaps this is related to : https://github.com/pyvista/pyvista/pull/2695  ?  Can you test the fix here with PBR?

**Comment by akaszynski** [comment]
If possible, please post your code here so we can reproduce.

You might be able to fix this by using the [cell_data_to_point_data](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSetFilters.cell_data_to_point_data.html) filter.

**Reply by danbodily** [reply]
I was on PyVista 0.37 -  When I upgraded to 0.38 this went away.  I suspect it was related to #2695.   Thank you!
